### ISSUE TITLE
Consider removing `_false` methods

### ISSUE THREAD

**Issue by izik1** [U007D/bool_ext#1]
Currently, the API surface contains the following methods:
```rust
as_option_false
as_result_false
do_false
expect_false
ok_false
ok_false_with
ok_or_err_false
ok_or_err_false_with
some_false
some_with_false
```
All of which have a corresponding `true` variant. At first blush, this seems reasonable, after all, it follows the API of `Option<T>` somewhat, however, booleans can be trivially inverted via:
```rust
!(boolean).method()

// or

use std::ops::Not;
boolean.not().method()
```

I'd argue that in the general case, it'd be far more likely for someone to write `some_bool.not().expect("Foo")` than `some_bool.expect_false("Foo")`. Additionally, I'd argue that the smaller API space (10 less methods) would make the API easier to understand and use due to it's smaller size.

ps: if neither option above is okay, the first one because it requires parens around the whole expression before it, the latter because of an extra import, I'd suggest adding a `fn BoolExt::invert(self) -> Self` function that just does `!self`.

**Comment by U007D** [comment]
The driver behind the `_false` cases was to make the resulting code more readable, based on the principle that code is written relatively few times but read relatively many times.  With that as a guiding principle, having a larger API surface is fine (so long as it continues to be zero runtime overhead).

In the [reddit thread](https://www.reddit.com/r/rust/comments/h0zuqj/announcing_bool_ext_a_more_complete_set_of/ful4ktm?utm_source=share&utm_medium=web2x) a suggestion was made to map the `false` names to `or`, making them more similar to the rest of `std`.  I'm curious to know whether a better harmony with existing convention would help instead.

**Comment by U007D** [comment]
Whew, I gave this a lot of thought, izik1, and after renaming all the `_false` methods to `_inverted` then trying prefix `and` and `or` as prefix, I came to the conclusion that your are probably right. :)  I've removed _false for all the methods except `do_false()`, which is now `or_do()` to correspond with `do_true()` becoming `and_do()`.
I expect "`do`" to be used as an `if`, `if/else` or even a standalone `else` statement, so I wanted those capabilities to be built in to the API.
Also gave credit/attribution to you in the README.md.  Thank you!
